src/offset.js

```diff
@@ -103,7 +103,7 @@ jQuery.fn.extend( {
 		rect = elem.getBoundingClientRect();
 
 		// Make sure element is not hidden (display: none)
-		if ( rect.width || rect.height ) {
+		if ( elem.style.display !== 'none' ) {
```

Comment:
What if its parent is `display; none`?

We would also need a few unit tests for this case to be sure that it works now and doesn't break in the future.